include/nbl/asset/IBuffer.h

```diff
@@ -18,6 +18,27 @@ class IBuffer : public core::IBuffer, public IDescriptor
 	public:
 		E_CATEGORY getTypeCategory() const override { return EC_BUFFER; }
 
+        enum E_USAGE_FLAGS : uint32_t
+        {
+            EUF_TRANSFER_SRC_BIT = 0x00000001,
+            EUF_TRANSFER_DST_BIT = 0x00000002,
+            EUF_UNIFORM_TEXEL_BUFFER_BIT = 0x00000004,
+            EUF_STORAGE_TEXEL_BUFFER_BIT = 0x00000008,
+            EUF_UNIFORM_BUFFER_BIT = 0x00000010,
+            EUF_STORAGE_BUFFER_BIT = 0x00000020,
+            EUF_INDEX_BUFFER_BIT = 0x00000040,
+            EUF_VERTEX_BUFFER_BIT = 0x00000080,
+            EUF_INDIRECT_BUFFER_BIT = 0x00000100
+        };
+
+        struct SCreationParams
+        {
+            core::bitflag<E_USAGE_FLAGS> usage;
+            E_SHARING_MODE sharingMode;
+            uint32_t queueFamilyIndexCount;
+            const uint32_t* queueFamilyIndices;
+        };
```

Comment:
@Crisspl wtf did you merge?

Reply:
assign @achalpandeyy , this needs to move IGPUBuffer

in ICPUBuffer and ICPUImage you only need to track `core::bitflag<E_USAGE_FLAGS> m_usage;`